src/daemons/GraphDaemon.cpp

```diff
@@ -166,33 +174,56 @@ int main(int argc, char *argv[]) {
     return EXIT_FAILURE;
   }
 
-  gServer->setPort(localhost.port);
-  gServer->setInterface(std::move(interface));
-  gServer->setReusePort(FLAGS_reuse_port);
-  gServer->setIdleTimeout(std::chrono::seconds(FLAGS_client_idle_timeout_secs));
-  gServer->setNumAcceptThreads(FLAGS_num_accept_threads);
-  gServer->setListenBacklog(FLAGS_listen_backlog);
-  if (FLAGS_enable_ssl || FLAGS_enable_graph_ssl) {
-    gServer->setSSLConfig(nebula::sslContextConfig());
-  }
-  setupThreadManager();
-
   // Setup the signal handlers
   status = setupSignalHandler();
   if (!status.ok()) {
     LOG(ERROR) << status;
     return EXIT_FAILURE;
   }
 
-  FLOG_INFO("Starting nebula-graphd on %s:%d\n", localhost.host.c_str(), localhost.port);
-  try {
-    gServer->serve();  // Blocking wait until shut down via gServer->stop()
-  } catch (const std::exception &e) {
-    FLOG_ERROR("Exception thrown while starting the RPC server: %s", e.what());
+  std::thread graph_thread([&] {
+    gServer->setPort(localhost.port);
+    gServer->setInterface(std::move(interface));
+    gServer->setReusePort(FLAGS_reuse_port);
+    gServer->setIdleTimeout(std::chrono::seconds(FLAGS_client_idle_timeout_secs));
+    gServer->setNumAcceptThreads(FLAGS_num_accept_threads);
+    gServer->setListenBacklog(FLAGS_listen_backlog);
+    if (FLAGS_enable_ssl || FLAGS_enable_graph_ssl) {
+      gServer->setSSLConfig(nebula::sslContextConfig());
+    }
+    setupThreadManager();
+    serverStatus_.store(STATUS_RUNNING);
+    FLOG_INFO("Starting nebula-graphd on %s:%d\n", localhost.host.c_str(), localhost.port);
+    try {
+      gServer->serve();  // Blocking wait until shut down via gServer->stop()
```

Comment:
I think you just need call `serve` in this thread.

Reply:
When I call `serve()`, the deamon will be blocked here, I don’t know how to ensure that the `stop` call is not a worker thread